From 1cb67f629ece047005073e01d50eef156f165c90 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 10 Jun 2019 14:16:06 +0300
Subject: [PATCH] [SW-0] pll: remove needless return values

The hl_pll_cfg() function does not return any meaningful value,
but just 0, so remove the return value at all.
Once we remove the return value from the hl_pll_cfg() function, the same
can be done for hl_pll_set() function, as no value it can propagate to
the caller.

Change-Id: I7199b87e0ad2591a6d593d2fdaf0fa44de1341c3
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index eb6ef92..2e9a124 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -50,7 +50,7 @@
 	regmap_writel(reg & ~HL_PLL_CFG_LOCK, cfg->base + HL_PLL_CFG);
 }
 
-static int hl_pll_cfg(struct hl_pll_cfg *cfg)
+static void hl_pll_cfg(struct hl_pll_cfg *cfg)
 {
 	hl_pll_reset(cfg, true);
 
@@ -67,8 +67,6 @@
 
 	/* Wait >500 (10us) reference cycles until PLL re-locks. */
 	hl_pll_udelay(HL_PLL_MAX_WAIT_TIME);
-
-	return 0;
 }
 
 static void hl_pll_clk_switch_to_ref(struct hl_pll_cfg *cfg)
@@ -178,13 +176,11 @@
 	regmap_writel(1, cfg->base + HL_PLL_LOCK_INTR);
 }
 
-static int hl_pll_set(struct hl_pll_cfg *cfg)
+static void hl_pll_set(struct hl_pll_cfg *cfg)
 {
-	int ret;
-
 	hl_pll_clk_switch_to_ref(cfg);
 
-	ret = hl_pll_cfg(cfg);
+	hl_pll_cfg(cfg);
 
 	hl_pll_div_init(cfg);
 
@@ -197,8 +193,6 @@
 	hl_pll_ack_int(cfg);
 
 	hl_pll_clk_switch_to_pll(cfg);
-
-	return ret;
 }
 
 u32 hl_pll_get_freq(u64 pll_base, enum hl_pll_div_out div)
